src/main/java/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package operationwithfruits;
+
+import core.basesyntax.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

This method returns `null` if the map doesn't contain a handler for the given operation. This could lead to a `NullPointerException` in `ShopServiceImpl` when `handler.handle()` is called. To make the code more robust, you should check if the handler exists and throw an exception if it's not found, for example: `throw new RuntimeException("No handler found for operation: " + operation);`.